packages/sanity/src/core/store/bundles/useBundleOperations.ts

```diff
@@ -59,19 +60,42 @@ export function useBundleOperations() {
         .filter(([_, value]) => value === undefined)
         .map(([key]) => key)
 
-      let clientOperation = client.patch(bundle._id).set(document)
+      let clientOperation = addOnClient.patch(bundle._id).set(document)
       if (unsetKeys.length) {
         clientOperation = clientOperation.unset(unsetKeys)
       }
 
       return clientOperation.commit()
     },
-    [client],
+    [addOnClient],
+  )
+
+  const handlePublishBundle = useCallback(
+    async (bundleId: string, bundleDocuments: SanityDocument[]) => {
+      if (!addOnClient) return null
+
+      const transaction = studioClient.transaction()
+      bundleDocuments.forEach((document) => {
+        const transactionDocument = omit(document, ['_version']) as SanityDocument
+
+        // update the published document with the bundle version
+        transaction.createOrReplace({
+          ...transactionDocument,
+          _id: getPublishedId(document._id, true),
+        })
```

Comment:
Tested using the actions api and versions are not supported yet. 
But I wonder if we could add here a check for the published revision id like in: https://github.com/sanity-io/sanity/blob/55ceee3298103ae473c99f06780ec9bdbe932bd3/packages/sanity/src/core/store/_legacy/document/document-pair/operations/publish.ts#L47-L53

Reply:
This link isn't working for me @pedrobonamin

Reply:
https://github.com/sanity-io/sanity/blob/corel/packages/sanity/src/core/store/_legacy/document/document-pair/operations/publish.ts#L47-L53